Description: Fix sphinx docs build
 When building docs with sphinx, the objects.register_all() code path is
 not necessarily called, and therefore objects.* is not necessarily
 imported before it is used. This throws an error while building the
 sample ChassisCollection response.
 .
 If objects.Chassis is not available, call register_all() before using
 it.
Author: Jim Rollenhagen <jim@jimrollenhagen.com>
Date: Tue, 22 Mar 2016 14:37:21 +0000 (+0000)
X-Git-Url: https://review.openstack.org/gitweb?p=openstack%2Fironic.git;a=commitdiff_plain;h=f2e0d93cbed2909ca3033ce8e20768ba4d3fd6b7
Change-Id: Ic0041085639f3d5a900ad54c2637c77aa467ba35
Closes-Bug: #1560508
Origin: upstream, https://review.openstack.org/#/c/295868/
Last-Update: 2016-03-22

diff --git a/ironic/api/controllers/v1/chassis.py b/ironic/api/controllers/v1/chassis.py
index 819278f..a14f96c 100644
--- a/ironic/api/controllers/v1/chassis.py
+++ b/ironic/api/controllers/v1/chassis.py
@@ -143,6 +143,9 @@ class ChassisCollection(collection.Collection):
 
     @classmethod
     def sample(cls):
+        # NOTE(jroll) hack for docs build, bug #1560508
+        if not hasattr(objects, 'Chassis'):
+            objects.register_all()
         sample = cls()
         sample.chassis = [Chassis.sample(expand=False)]
         return sample
